test/test_core.h

```diff
@@ -69,11 +69,11 @@ device_t device_type(size_t *platform, size_t *device) {
 }
 
 #define TINY_DNN_GET_DEVICE_AND_PLATFORM       \
-    size_t cl_platform = 0, cl_device = 0; \
+    int cl_platform = 0, cl_device = 0; \
```

Comment:
I've just reverted the change in here. I'll send another PR for this in future.

The reason why I changed the type : 2nd and 3rd parameters of `Device::Device` in `tiny_dnn/core/framework/device.fwd.h` is `int`, maybe I should have used `size_t` -> `int` casts.